site/src/components/CopyableValue/CopyableValue.tsx

```diff
@@ -1,35 +1,71 @@
-import Tooltip, { type TooltipProps } from "@mui/material/Tooltip";
+import {
+	Tooltip,
+	TooltipContent,
+	TooltipProvider,
+	TooltipTrigger,
+} from "components/Tooltip/Tooltip";
 import { useClickable } from "hooks/useClickable";
 import { useClipboard } from "hooks/useClipboard";
-import type { FC, HTMLAttributes } from "react";
+import { type FC, type HTMLAttributes, useRef, useState } from "react";
+import { cn } from "utils/cn";
+
+type TooltipSide = "top" | "right" | "bottom" | "left";
 
 interface CopyableValueProps extends HTMLAttributes<HTMLSpanElement> {
 	value: string;
-	placement?: TooltipProps["placement"];
-	PopperProps?: TooltipProps["PopperProps"];
+	side?: TooltipSide;
 }
 
 export const CopyableValue: FC<CopyableValueProps> = ({
 	value,
-	placement = "bottom-start",
-	PopperProps,
+	side = "bottom",
 	children,
+	className,
 	...attrs
 }) => {
 	const { showCopiedSuccess, copyToClipboard } = useClipboard();
+	const [tooltipOpen, setTooltipOpen] = useState(false);
+	const [showCopiedText, setShowCopiedText] = useState(false);
+	const prevCopiedRef = useRef(false);
+	const timeoutRef = useRef<number | undefined>(undefined);
+
 	const clickableProps = useClickable<HTMLSpanElement>(() => {
 		copyToClipboard(value);
+		setTooltipOpen(true);
 	});
 
+	if (showCopiedSuccess !== prevCopiedRef.current) {
+		prevCopiedRef.current = showCopiedSuccess;
+
+		if (showCopiedSuccess) {
+			setShowCopiedText(true);
```

Comment:
why have both `showCopiedSuccess` and `showCopiedText`? this just feels like duplicated state with messy synchronization code